lib/Doctrine/ORM/PersistentCollection.php

```diff
@@ -83,18 +78,17 @@ final class PersistentCollection extends AbstractLazyCollection implements Selec
     /**
      * Creates a new persistent collection.
      *
-     * @param EntityManagerInterface $em    The EntityManager the collection will be associated with.
-     * @param ClassMetadata          $class The class descriptor of the entity type of this collection.
+     * @param EntityManagerInterface $em        The EntityManager the collection will be associated with.
+     * @param ClassMetadata          $typeClass The class descriptor of the entity type of this collection.
      * @psalm-param Collection<TKey, T>&Selectable<TKey, T> $collection The collection elements.
      */
     public function __construct(
         EntityManagerInterface $em,
-        ClassMetadata $class,
+        private readonly ClassMetadata|null $typeClass,
         Collection $collection,
```

Comment:
`$em` and `$collection` could also use constructor promotion.

Reply:
I don't think so: `$em` is not supposed to be passed as `null`, but can become `null` on unserialization. `$collection` is a property inherited from the parent.